src/main_menu.cpp

```diff
@@ -619,13 +619,16 @@ bool game::opening_screen()
             }
         } else if (layer == 3) {
             bool available = false;
-            std::string wn = world_generator->all_worldnames[sel3];
 
-            if ( (wn != "TUTORIAL" && wn != "DEFENSE") && world_generator->world_need_lua_build(wn) ) {
-                layer = 2;
-                sel1 = 2;
-                continue;
+            std::string wn;
```

Comment:
Sooo... where is this variable written? Currently it crashes because this variable is always empty, and `world_need_lua_build` doesn't like being called with a non-existing world name. The function should probably check that the given world name is valid before accessing it.